### ISSUE TITLE
Module not upgrading agents on Windows

### ISSUE THREAD

**Issue by mike406** [wazuh/wazuh-puppet#374]
For Windows computers that already have a previous version of Wazuh agent installed, the agents are not automatically upgraded when the version is different. The new MSI shows up in C:\Temp but is not actually installed/upgraded over the old installation.

Changing ensure to use the MSI version resolves this problem. I hardcoded 4.1.5 but a new variable should be established that uses the MSI version number.
```
      -> package { $agent_package_name:
        ensure          => "4.1.5",
        provider        => 'windows',
        source          => "${download_path}\\wazuh-agent-${agent_package_version}.msi",
        install_options => [
          '/q',
          "WAZUH_MANAGER=${wazuh_reporting_endpoint}",
          "WAZUH_PROTOCOL=${ossec_protocol}",
        ],
      }
```

```
Info: Applying configuration version '1628613757'
Notice: /Stage[main]/Wazuh::Agent/File[wazuh-agent]/ensure: defined content as '{md5}50455229db051712fef18eb7cda1ce65'
Notice: /Stage[main]/Wazuh::Agent/Package[Wazuh Agent]/ensure: ensure changed '4.0.4' to '4.1.5'
```

**Comment by vcerenu** [comment]
The incident that the user had was replicated in a test environment. Different tests were carried out for the modification and among them it was taken as a partial solution to send the Wazuh version as a parameter

```
      -> package { $agent_package_name:
        ensure          => "${agent_package_version}",
        provider        => 'windows',
        source          => "${download_path}\\wazuh-agent-${agent_package_version}.msi",
        install_options => [
          '/q',
          "WAZUH_MANAGER=${wazuh_reporting_endpoint}",
          "WAZUH_PROTOCOL=${ossec_protocol}",
        ],
      }
```
Operation tests are carried out and if they are completed without problems, this change will be added to the repository

**Comment by vcerenu** [comment]
Hi Mike 

I proceed to close this Issue since this change was implemented for version [v4.2.4](https://github.com/wazuh/wazuh-puppet/tree/v4.2.4)
